src/line-top-index.js

```diff
@@ -69,53 +69,62 @@ export default class LineTopIndex {
     let oldEnd = traverse(start, oldExtent)
     let newEnd = traverse(start, newExtent)
 
+    let isInsertion = isZero(oldExtent)
     let startNode = this.iterator.insertNode(start)
-    let endNode = this.iterator.insertNode(oldEnd)
+    let endNode = this.iterator.insertNode(oldEnd, !isInsertion)
```

Comment:
One thing I ended up doing in the patch was to actually pass the `startNode`, and then only return an existing node if it doesn't match the given node.